src/Entity/ItemId.php

```diff
@@ -24,7 +25,7 @@ class ItemId extends EntityId implements Int32EntityId {
 	 */
 	public function __construct( $idSerialization ) {
 		$this->assertValidIdFormat( $idSerialization );
```

Comment:
I'm not sure if you already saw the latest version of this but validating the two parts completely separately is kind of a chicken-egg-problem since the splitting method also wants to validate the serialization. EntityId now looks at the entire string to validate the prefix and also makes sure that there's *something* at the end that does not contain colons. The last part is then validated in the subclass as it should be.